app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,80 @@
+module Users
+  class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+    def google_oauth2
+      if !CheckOmniauth.verified?(auth)
+        when_social_profile_not_verified
+      else
+        handle_callback
+      end
+    end
+
+    def facebook
+      if !CheckOmniauth.verified?(auth)
+        when_social_profile_not_verified
```

Comment:
not really DRY to repeat one check in the every single action, WDYT about moving this check to callback before_action?